common/test/acceptance/tests/studio/test_studio_container.py

```diff
@@ -1136,3 +1137,230 @@ def test_common_problem_types_tab(self):
+        elif operation == 'undo_move':
+            self.container.click_undo_move_link()
+            self.container.verify_confirmation_message(
+                self.message_undo.format(display_name=self.source_component_display_name)
+            )
+
+        unit_page = ContainerPage(self.browser, None)
+        components = unit_page.displayed_children
+        self.assertEqual(
+            [component.name for component in components],
+            component_display_names_after_operation
+        )
+
+    def test_move(self):
+        """
+        Test if we can move a component successfully.
+
+        Given I am a staff user
+        When I go to unit page in first section
+        Then I open the move modal
+        Then I navigate to unit in second section from within move modal
+        Then I see move button is enabled
+        Then I click on the move button
+        Then I see move operation successful message
+        When I go to unit page in second section
+        Then I see moved component there
+        """
+        unit_page = self.go_to_unit_page(unit_name='Test Unit 1')
+        components = unit_page.displayed_children
+        self.assertEqual(len(components), 2)
+
+        self.verify_move_opertions(
+            unit_page=unit_page,
+            source_component=components[0],
+            operation='move',
+            component_display_names_after_operation=['HTML 21', 'HTML 22', 'HTML 11']
+        )
+
+    def test_undo_move(self):
+        """
+        Test if we can undo move a component successfully.
+
+        Given I am a staff user
+        When I go to unit page in first section
+        Then I open the move modal
+        Then I click on the move button
+        Then I see move operation successful message
+        When I clicked on undo move link
+        Then I verified that undo move operation is successful
+        """
+        unit_page = self.go_to_unit_page(unit_name='Test Unit 1')
+        components = unit_page.displayed_children
+        self.assertEqual(len(components), 2)
+
+        self.verify_move_opertions(
+            unit_page=unit_page,
+            source_component=components[0],
+            operation='undo_move',
+            component_display_names_after_operation=['HTML 11', 'HTML 12']
+        )
+
+    def test_content_experiment(self):
+        """
+        Test if we can move a component of content experiment successfully.
+
+        Given I am a staff user
+        When I go to content experiment page
+        Then I open the move modal
+        Then I navigate to unit in second section from within move modal
+        Then I see move button is enabled
+        Then I click on the move button
+        Then I see move operation successful message
+        When I go to unit page in second section
+        Then I see moved component there
+        When I undo move a component
+        Then I verified that undo move operation is successful
```

Comment:
@muzaffaryousaf we are doing move initially and then checking confirmation message, if it is only move operation, we are checking take me there link otherwise we do undo move and check it's confirmation as well.